api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -67,6 +75,14 @@ public void setUserDAO(UserDAO dao) {
 		this.dao = dao;
 	}
 	
+	/**
+	 * @return the validTime for which the password reset activation key will be valid
+	 */
+	private int getValidTime() {
+		//if valid time is less that a minute or greater than 12hrs reset valid time to 1 minutes else set it to the required time.
+		return (validTime < MIN_VALID_TIME) || (validTime > MAX_VALID_TIME) ? DEFAULT_VALID_TIME : validTime;
```

Comment:
What is happening here? The valid time needs to be read from a global property that you will have to add, if the value is less or more than the min and max then you use the default.

Reply:
Recall on our call I said you need to use a global property to let an admin define their preferred valid time, and that way also in your test you set the global property value to 1min for testing purposes so that the test doesn't have to last for the default 10min, do you know what a global property is in OpenMRS and how to use it?